packages/gatsby-transformer-code-examples/gatsby-node.js

```diff
@@ -0,0 +1,74 @@
+const createSchemaCustomization = ({ actions, schema }) => {
+  const { createTypes } = actions;
+
+  const typeDefs = [
+    schema.buildObjectType({
+      name: 'CodeExample',
+      fields: {
+        name: 'String!',
+        extension: 'String!',
+        absolutePath: 'String!',
+        content: 'String!',
+      },
+      interfaces: ['Node'],
+      extensions: {
+        infer: false,
+      },
+    }),
+  ];
+
+  createTypes(typeDefs);
+};
+
+async function onCreateNode({
+  node,
+  actions,
+  loadNodeContent,
+  createNodeId,
+  createContentDigest,
+}) {
+  const validExtensions = [
+    'application/javascript',
+    'text/x-java-source',
+    'application/octet-stream',
+    'application/json',
+    'application/x-httpd-php',
+    'application/x-sh',
+    'video/mp2t',
```

Comment:
Maybe add an explanatory comment that this is typescript, which does not (yet) have an own media type.